plugins/newsql/newsql_evbuffer.c

```diff
@@ -356,10 +357,23 @@ static void process_dbinfo_int(struct newsql_appdata_evbuffer *appdata, struct e
     struct newsqlheader hdr = {0};
     hdr.type = htonl(RESPONSE_HEADER__DBINFO_RESPONSE);
     hdr.length = htonl(len);
-    uint8_t out[len];
+    uint8_t *out = malloc(len);
     cdb2__dbinforesponse__pack(&response, out);
     evbuffer_add(buf, &hdr, sizeof(hdr));
     evbuffer_add(buf, out, len);
```

Comment:
Should we still send if can't decode?

Reply:
Where does it get freed?

Reply:
What would that do? Client will fail to decode and crash.

Reply:
Right now it looks like the code can potentially send the message first and then crash

Reply:
Ah ok ic

Reply:
We write after this routine returns (which we won't, if our buffer is trashed):
https://github.com/bloomberg/comdb2/blob/master/plugins/newsql/newsql_evbuffer.c#L367-L368

Reply:
Ah I was thinking about bufferevents which I guess write for you